pkg/tests/ossm/smoke_test.go

```diff
@@ -19,32 +19,101 @@ import (
 	"fmt"
 	"strings"
 	"testing"
+	"time"
 
 	"github.com/maistra/maistra-test-tool/pkg/app"
 	"github.com/maistra/maistra-test-tool/pkg/util/check/assert"
+	"github.com/maistra/maistra-test-tool/pkg/util/env"
 	"github.com/maistra/maistra-test-tool/pkg/util/oc"
 	"github.com/maistra/maistra-test-tool/pkg/util/pod"
+	"github.com/maistra/maistra-test-tool/pkg/util/retry"
 	"github.com/maistra/maistra-test-tool/pkg/util/shell"
+	"github.com/maistra/maistra-test-tool/pkg/util/test"
 	. "github.com/maistra/maistra-test-tool/pkg/util/test"
+	"github.com/maistra/maistra-test-tool/pkg/util/version"
 )
 
-func TestBookinfoInjection(t *testing.T) {
-	NewTest(t).Id("A2").Groups(ARM, Full, Smoke, InterOp, Disconnected).Run(func(t TestHelper) {
+var (
+	VERSIONS = []*version.Version{
+		&version.SMCP_2_0,
+		&version.SMCP_2_1,
+		&version.SMCP_2_2,
+		&version.SMCP_2_3,
+		&version.SMCP_2_4,
+	}
+)
+
+type PodTime struct {
+	PodName            string
+	StartedAt          time.Time
+	LastTransitionTime time.Time
+}
+
+func TestSmoke(t *testing.T) {
+	NewTest(t).Groups(ARM, Full, Smoke, InterOp, Disconnected).Run(func(t TestHelper) {
+		t.Log("Smoke Test for SMCP: deploy, upgrade, bookinfo and uninstall")
 		ns := "bookinfo"
 
 		t.Cleanup(func() {
 			app.Uninstall(t, app.Bookinfo(ns), app.SleepNoSidecar(ns))
+			oc.RecreateNamespace(t, meshNamespace)
 		})
 
-		DeployControlPlane(t)
+		toVersion := env.GetSMCPVersion()
+		fromVersion := getPreviousVersion(toVersion)
+
+		oc.RecreateNamespace(t, meshNamespace)
 
-		t.LogStep("Install bookinfo pods with sidecar and sleep pod without sidecar")
+		t.LogStep("Install bookinfo pods and sleep pod")
 		app.InstallAndWaitReady(t, app.Bookinfo(ns), app.SleepNoSidecar(ns))
 
-		t.LogStep("Check whether sidecar is injected in all bookinfo pods")
-		assertSidecarInjectedInAllBookinfoPods(t, ns)
+		t.NewSubTest(fmt.Sprintf("install bookinfo with smcp %s", fromVersion)).Run(func(t TestHelper) {
+
+			t.LogStepf("Create SMCP %s and verify it becomes ready", fromVersion)
+			assertSMCPDeploysAndIsReady(t, fromVersion)
+
+			t.LogStep("Restart all pods to verify proxy is injected in all pods of Bookinfo")
+			oc.RestartAllPods(t, ns)
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				assertSidecarInjectedInAllBookinfoPods(t, ns)
+			})
+
+			t.LogStep("Check if bookinfo productpage is running through the Proxy")
+			assertTrafficFlowsThroughProxy(t, ns)
+
+			t.LogStep("verify proxy startup time. Expected to be less than 10 seconds")
+			t.Log("Jira related: https://issues.redhat.com/browse/OSSM-3586")
+			t.Log("From proxy json , verify the time between status.containerStatuses.state.running.startedAt and status.conditions[type=Ready].lastTransitionTime")
+			t.Log("The proxy startup time should be less than 10 seconds for ratings pod")
+			validateStartUpProxyTime(t, ns)
```

Comment:
```suggestion
			assertProxiesReadyInLessThan10Seconds(t, ns)
```

(or pass 10s as a parameter and rename the function to `assertProxiesReadyIn()`).